src/app/children/attendance/add-day-attendance/add-day-attendance.component.spec.ts

```diff
@@ -28,8 +30,9 @@ describe('AddDayAttendanceComponent', () => {
         MatIconModule, MatProgressBarModule, NoopAnimationsModule,
         EntityModule],
       providers: [
-        {provide: ChildrenService, useClass: ChildrenService},
+        ChildrenService,
```

Comment:
we should mock the ChildrenService to make it better testable and avoid having to provide all those dependencies of ChildrenService in these tests. I started refactoring for one test: https://github.com/Aam-Digital/ndb-core/pull/306/commits/60d838ef1211c45f69d8686d179303f2b8c51415